From 8032f80ebf4ea97aa3e2759e7fa1c51c9444c586 Mon Sep 17 00:00:00 2001
From: jtheertha <jtheertha@75f.io>
Date: Thu, 23 Jun 2022 11:41:11 +0000
Subject: [PATCH 28/50] Merged PR 4225: Number formatting for numbers

Number formatting for numbers in Alerts and emr status message

Related work items: #12436
---
 .../java/a75f/io/alerts/AlertFormatter.java   |  3 ++
 .../a75f/io/api/haystack/util/StringUtil.java | 28 +++++++++++++++++++
 .../io/logic/bo/building/erm/EmrEquip.java    |  5 ++--
 .../io/logic/bo/building/erm/EmrProfile.java  |  4 +--
 4 files changed, 36 insertions(+), 4 deletions(-)
 create mode 100644 haystack/src/main/java/a75f/io/api/haystack/util/StringUtil.java

diff --git a/alerts/src/main/java/a75f/io/alerts/AlertFormatter.java b/alerts/src/main/java/a75f/io/alerts/AlertFormatter.java
index a302b9a4b..df43c3f1d 100644
--- a/alerts/src/main/java/a75f/io/alerts/AlertFormatter.java
+++ b/alerts/src/main/java/a75f/io/alerts/AlertFormatter.java
@@ -7,6 +7,7 @@ import java.util.StringTokenizer;
 
 import a75f.io.api.haystack.CCUHsApi;
 import a75f.io.api.haystack.Point;
+import a75f.io.api.haystack.util.StringUtil;
 import a75f.io.logger.CcuLog;
 
 /***
@@ -58,6 +59,7 @@ public class AlertFormatter
                 }
             }
         }
+        message = StringUtil.processMessageForNumberFormatting(message);
         CcuLog.d("CCU_ALERTS","  Alert Formatted Message "+message);
         return message;
     }
@@ -71,6 +73,7 @@ public class AlertFormatter
                 message = replaceToken(message, token, def, pointId);
             }
         }
+        message = StringUtil.processMessageForNumberFormatting(message);
         CcuLog.d("CCU_ALERTS","  Alert Formatted Message "+message);
         return message;
     }
diff --git a/haystack/src/main/java/a75f/io/api/haystack/util/StringUtil.java b/haystack/src/main/java/a75f/io/api/haystack/util/StringUtil.java
new file mode 100644
index 000000000..9d7d4f204
--- /dev/null
+++ b/haystack/src/main/java/a75f/io/api/haystack/util/StringUtil.java
@@ -0,0 +1,28 @@
+package a75f.io.api.haystack.util;
+
+public class StringUtil {
+
+    private StringUtil(){
+
+    }
+    public static String processMessageForNumberFormatting(String message){
+        String space = " ";
+        String[] messageWords = message.split(space);
+        StringBuilder modifiedMessage = new StringBuilder();
+        for (String messageWord : messageWords){
+            modifiedMessage.append(space);
+            if(messageWord.matches("-?\\d+(\\.\\d+)?")){
+                if(messageWord.contains(".")) {
+                    modifiedMessage.append(String.format("%,.2f", Double.parseDouble(messageWord)));
+                }
+                else {
+                    modifiedMessage.append(String.format("%,d", Long.parseLong(messageWord)));
+                }
+            }
+            else{
+                modifiedMessage.append(messageWord);
+            }
+        }
+        return modifiedMessage.toString().trim();
+    }
+}
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrEquip.java b/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrEquip.java
index 9f2c01134..ecc503692 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrEquip.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrEquip.java
@@ -7,6 +7,7 @@ import a75f.io.api.haystack.Equip;
 import a75f.io.api.haystack.Kind;
 import a75f.io.api.haystack.Point;
 import a75f.io.api.haystack.Tags;
+import a75f.io.api.haystack.util.StringUtil;
 import a75f.io.logic.bo.building.definitions.Port;
 import a75f.io.logic.bo.building.definitions.ProfileType;
 import a75f.io.logic.bo.building.heartbeat.HeartBeat;
@@ -126,8 +127,8 @@ public class EmrEquip
     
     public void setEquipStatus(String status)
     {
-        hayStack.writeDefaultVal("point and status and message and equipRef == \""+equipRef+"\"", status);
-        
+        hayStack.writeDefaultVal("point and status and message and equipRef == \""+equipRef+"\"",
+                StringUtil.processMessageForNumberFormatting(status));
     }
     
     public void setHisVal(String query, double val)
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrProfile.java b/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrProfile.java
index 392bfcda5..76060b371 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrProfile.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/erm/EmrProfile.java
@@ -75,7 +75,7 @@ public class EmrProfile extends ZoneProfile
         int timeDiffMins = (int) (reading1.getDate().getTime() - reading2.getDate().getTime())/(60*1000);
         if (timeDiffMins < 1) {
             double curRate = emrEquip.getHisVal("current and rate");
-            emrEquip.setEquipStatus("Total Energy Consumed "+reading1.getVal()+" kWh "+" Current Rate "+curRate+"KW");
+            emrEquip.setEquipStatus("Total Energy Consumed "+reading1.getVal()+" kWh"+" Current Rate "+curRate+" KW");
             return;
         }
         double readingDiff = 100 * (reading1.getVal() - reading2.getVal());
@@ -85,7 +85,7 @@ public class EmrProfile extends ZoneProfile
         ratekWh = Math.round(ratekWh * 100)/100;
         
         emrEquip.setHisVal("current and rate", ratekWh);
-        emrEquip.setEquipStatus("Total Energy Consumed "+reading1.getVal()+" kWh "+" Current Rate "+ratekWh+"KW");
+        emrEquip.setEquipStatus("Total Energy Consumed "+reading1.getVal()+" kWh "+" Current Rate "+ratekWh+" KW");
     
         CcuLog.d(L.TAG_CCU_ZONE, "EmrProfile, Total Energy Consumed "+reading1.getVal()+" currentRate "+ratekWh);
         
-- 
2.31.0.windows.1

